copier/main.py

```diff
@@ -669,41 +690,30 @@ def _render_symlink(self, src_abspath: Path) -> None:
             return
 
         if not self.pretend:
+            dst_abspath = self.subproject.local_abspath / dst_relpath
             # symlink_to doesn't overwrite existing files, so delete it first
             if dst_abspath.is_symlink() or dst_abspath.exists():
                 dst_abspath.unlink()
+            dst_abspath.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
If you add this line also to `_render_file` maybe you can remove `_render_folder`?

Reply:
No, i don't think so because we need to be able to render empty directories and `_render_{file,symlink}()` only get called for file paths but not directory paths.

Reply:
Then this line can be removed from here?

Reply:
No – although it seems like it at first glance – because there is an edge case that is covered by the new test `tests/test_exclude.py::test_exclude_wildcard_negate_nested_file`:

There, we have a symlink `foo/bar/keep-me-symlink.txt` that targets the file `foo/keep-me.txt`. At the same time, we exclude everything via `*` except those two files via negative excludes (which wasn't working prior to this PR). This means, the directories `foo/` and `foo/bar/` are excluded according to the exclude patterns, but `Worker._render_file()` is called for `foo/keep-me.txt` and `Worker._render_symlink()` is called for `foo/bar/keep-me-symlink.txt` without any prior call of `Worker._render_folder()` (for `foo/` or `foo/bar/`). Thus, it's important for `Worker._render_{file,symlink}()` to also create their parent directories even though it seems unnecessary.

Reply:
> If you add this line also to `_render_file` maybe you can remove `_render_folder`?

By the way, `_render_file` already has this line:

https://github.com/copier-org/copier/blob/7209b3b35ea82672f3cf9d121ef95c551c7355a2/copier/main.py#L635